client/my-sites/checkout/src/components/refund-policies.tsx

```diff
@@ -78,13 +78,21 @@ export function getRefundPolicies( cart: ResponseCart ): RefundPolicy[] {
 		}
 
 		if ( isDomainRegistration( product ) ) {
+			if ( has100YearDomain( cart ) ) {
+				return RefundPolicy.GenericCentennial;
+			}
+
 			if ( isRenewal( product ) ) {
 				return RefundPolicy.DomainNameRenewal;
 			}
 			return RefundPolicy.DomainNameRegistration;
 		}
 
 		if ( isDomainTransfer( product ) ) {
+			if ( has100YearDomain( cart ) ) {
+				return RefundPolicy.GenericCentennial;
+			}
+
```

Comment:
I am not sure how to test the transfer flow since it asks for a domain that's unlocked and the transfer code before reaching the checkout screen.